ios/NativeSigner/Screens/MainScreenContainer.swift

```diff
@@ -24,74 +24,74 @@ struct MainScreenContainer: View {
+                                doWipe: data.wipe,
+                                alertShow: {data.alertShow = true},
+                                increment: { seedName, details in
+                                    let seedPhrase = data.getSeed(seedName: seedName)
+                                    if seedPhrase != "" {
+                                        data.pushButton(action: .increment, details: "1", seedPhrase: seedPhrase)
+                                    }
+                                }
+                            )
+                            Spacer()
+                        }
+                        ModalSelector(
+                            modalData: data.actionResult.modalData,
                             alert: data.alert,
+                            alertShow: {data.alertShow = true},
                             pushButton: {action, details, seedPhrase in data.pushButton(action: action, details: details, seedPhrase: seedPhrase)},
-                            getSeed: {seedName in return data.getSeed(seedName: seedName)},
-                            doJailbreak: data.jailbreak,
-                            pathCheck: {seed, path, network in
-                                return substratePathCheck(seedName: seed, path: path, network: network, dbname: data.dbName)
-                            },
+                            removeSeed: { seedName in data.removeSeed(seedName: seedName)},
+                            restoreSeed: {seedName, seedPhrase, createSeedKeys in data.restoreSeed(seedName: seedName, seedPhrase: seedPhrase, createRoots: createSeedKeys)},
                             createAddress: {path, seedName in data.createAddress(path: path, seedName: seedName)},
-                            checkSeedCollision: {seedName in return data.checkSeedCollision(seedName: seedName)},
-                            restoreSeed: {seedName, seedPhrase, createRoots in data.restoreSeed(seedName: seedName, seedPhrase: seedPhrase, createRoots: createRoots)},
-                            sign: {seedName, comment in data.sign(seedName: seedName, comment: comment)},
-                            doWipe: data.wipe,
-                            alertShow: {data.alertShow = true},
-                            increment: { seedName, details in
-                                let seedPhrase = data.getSeed(seedName: seedName)
-                                if seedPhrase != "" {
-                                    data.pushButton(action: .increment, details: "1", seedPhrase: seedPhrase)
-                                }
-                            }
+                            getSeedForBackup: {seedName in return data.getSeed(seedName: seedName, backup: true)},
+                            sign: {seedName, comment in data.sign(seedName: seedName, comment: comment)}
+                        )
+                        AlertSelector(
```

Comment:
Just curious, any reason not to pass .environmentObject(data) to AlertSelector?

Reply:
> there is no environment object anymore

I'm curious why it's not used. Is it a bad practice?

Reply:
In general, no.

But in this particular case, we've moved truth source onto particular screen and instead of passing the whole object, we pass only item-specific data and closures. This makes it easier to use screen previews and in general test things separately, since now there is no need to mock the whole data, only screen-specific part of it.

Global variables make sense if it's indeed something small that's used in many screens, but even network detection alert is does not fit to this description.